### ISSUE TITLE
Fine-Tuning problems

### ISSUE THREAD

**Issue by xxxpd456** [microsoft/aurora#112]
Thank you for your excellent work on the Aurora model — it's been incredibly promising for high-resolution global weather forecasting.

I’m currently attempting to fine-tune Aurora on a custom ERA5 dataset (e.g., adding new surface variables like 2m dewpoint and specific pressure levels such as 950 hPa). While I have gone through the available documentation and supplementary materials, I’m still unclear on the full recommended procedure for implementing fine-tuning in practice.

Could you please provide:

A complete, minimal working example (preferably Python script) for fine-tuning Aurora using a local dataset;

Any official script or reference implementation used for fine-tuning in the paper or experiments.

If such an example already exists, I’d appreciate a link or guidance on where to find it.Thanks again for your help and for open-sourcing such an impactful model!

**Comment by sergioleonluis** [comment]
Hey @xxxpd456 and @wesselb 

I have started the fine-tuning AURORA model using ERA5 data. 
I am not an expert in training AI models, so my scheme may not be correct (maybe, someone can guide us better) 
I describe you the steps I have done:

1) I have downloaded the ERA5 hourly data at surface and at different pressure levels (50,100,...) for all 2018.
2) My script selects a random date T0=(2018, month, day, hour) and I build 3 batch (T0, T0+6h,T0+12h).

3) I proceed to train Aurora as follows:

 optimizer.zero_grad()
 pred = model.forward(batch)
 loss = weighted_loss1(pred, batchT6, surface_weights,atmos_weights)
 pred2 = model.forward(pred)
 loss2 = weighted_loss2(pred2, batchT12, surface_weights, atmos_weights)
 loss = (loss+loss2)/2
    
The function loss and the weights are defined in the supplementary information in the Aurora article.

4) I repeat step 2-3, in a loop of 200 repetitions and, after that, I save my trained model.
Note: this scheme corresponds to Short-lead-time-fine-tunning

Like you, @xxxpd456,  I have my doubts whether I am proceeding correctly in the Aurora training mode. 
As I use CPU which slows down getting results I only do 200 reps at the moment.

Regards,
Sergio

**Comment by sergioleonluis** [comment]
Thank you very much for your feedback @wesselb 

I will follow your advice and reduce my script to a single forward. So, the processing time will be faster and I will be able to increase the number of iterations.

I am still looking for some GPU to go faster :)

**Comment by wesselb** [comment]
Best of luck! Hope you can get things working. :)
